Controller/CRUDController.php

```diff
@@ -1080,7 +1080,9 @@ protected function redirectTo($object)
 
         if (!$url) {
             foreach (array('edit', 'show') as $route) {
-                if ($this->admin->hasRoute($route) && $this->admin->isGranted(strtoupper($route), $object)) {
+                if ($this->admin->hasRoute($route) 
+                    && $this->admin->isGranted(strtoupper(str_replace('show', 'view', $route)), $object)
```

Comment:
I think the `checkAccess` method has built-in logic that should work here. This looks too fragile.